axiom/datasets.go

```diff
@@ -560,6 +571,48 @@ func (s *DatasetsService) IngestEvents(ctx context.Context, id string, opts Inge
 	return &res, nil
 }
 
+// IngestChannel ingests events from a channel into the dataset identified by
+// its id.
+// Restrictions for field names (JSON object keys) can be reviewed here:
+// https://www.axiom.co/docs/usage/field-restrictions.
+func (s *DatasetsService) IngestChannel(ctx context.Context, id string, events <-chan Event, opts IngestOptions) (*IngestStatus, error) {
```

Comment:
I think this method could use multiple cleanups:

1. It does not allow any control over the size of the buffer. I don't know if that is something users **really** need but if they are using channels it feels like they have a more complex use case that needs more control over ingestions, as well.
2. Instead of saving into a buffer we might profit from opening a connection and writing directly to the request body (using an `io.Pipe`). E.g. the whole `IngestEvents` method but ranging over the channel instead of the supplied slice of events.
3. Code seems a bit repetitive, could make the flush logic a reusable function.
4. It is logging error instead of returning them. Big no no.

Reply:
Re 4.: Oops, yeah that log should not be in there.